bulkclient/src/main/java/au/csiro/pathling/export/ws/BulkExportRequest.java

```diff
@@ -0,0 +1,278 @@
+import au.csiro.pathling.export.fhir.Parameters.Parameter;
+import au.csiro.pathling.export.fhir.Reference;
+import au.csiro.pathling.export.utils.WebUtils;
+import au.csiro.pathling.utilities.Lists;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import lombok.Builder;
+import lombok.Value;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.client.utils.URIBuilder;
+
+/**
+ * Represents a request to initiate a bulk export operation.
+ *
+ * @see <a href="https://hl7.org/fhir/uv/bulkdata/export.html#query-parameters">FHIR Bulk Export
+ * Request Query Parameters</a>
+ */
+@Value
+@Builder()
+public class BulkExportRequest {
+
+  /**
+   * The level of the export operation.
+   */
+  public interface Level {
+
+    /**
+     * The path to the export operation corresponding to this level.
+     *
+     * @return the path to the export operation.
+     */
+    @Nonnull
+    String getPath();
+
+    /**
+     * Whether this level supports patient-specific exports.
+     *
+     * @return true if patient-specific exports are supported.
+     */
+    boolean isPatientSupported();
+  }
+
+  /**
+   * Represents the system level of the export operation.
+   */
+  @Value
+  public static class SystemLevel implements Level {
+
+    @Nonnull
+    @Override
+    public String getPath() {
+      return "$export";
+    }
+
+    @Override
+    public boolean isPatientSupported() {
+      return false;
+    }
+  }
+
+  /**
+   * Represents the patient level of the export operation.
+   */
+  @Value
+  public static class PatientLevel implements Level {
+
+
+    @Nonnull
+    @Override
+    public String getPath() {
+      return "Patient/$export";
+    }
+
+    @Override
+    public boolean isPatientSupported() {
+      return true;
+    }
+  }
+
+  /**
+   * Represents the group level of the export operation.
+   */
+  @Value
+  public static class GroupLevel implements Level {
+
+    /**
+     * The ID of the group.
+     */
+    @Nonnull
+    String id;
+
+    @Nonnull
+    @Override
+    public String getPath() {
+      return String.format("Group/%s/$export", id);
+    }
+
+    @Override
+    public boolean isPatientSupported() {
+      return true;
+    }
+  }
+
+  /**
+   * The level of the export operation. The default is {@link SystemLevel}.
+   */
+  @Nonnull
+  @Builder.Default
+  Level level = new SystemLevel();
+
+  /**
+   * The format of the output. The value of the '_outputFormat' query parameter.
+   */
+  @Nullable
+  @Builder.Default
+  String _outputFormat = null;
```

Comment:
What happens if I request something other than NDJSON, and receive it?

Do we have assumptions elsewhere that will be broken by this?

Reply:
Not at the moment. The exported files are 'blackboxes' we just save what the server provides without any interpretation. 
The only potentially correlated thing is the:
BulkExportClient:
  /**
   * The extension to use for the output files.
   */
  @Nonnull
  @Builder.Default
  String outputExtension = "ndjson";
  
  
  This will need to be manually adjusted to the actual format.
  
  This may become a bit more complicated if we add support for downloading 'error' URLs from the response as these should be JSON operation outcomes.

Reply:
I think we can leave this as is for now, as NDJSON is the only current format.